lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -443,6 +443,13 @@ static std::string getSubModuleName(Operation *oldOp) {
       subModuleName += "_seq";
     else
       subModuleName += "_fifo";
+
+    if (bufferOp.initValues().hasValue()) {
+      subModuleName += "_init";
+      for (auto e : *bufferOp.initValues())
+        subModuleName +=
+            "_" + std::to_string(e.dyn_cast<IntegerAttr>().getInt());
```

Comment:
Won't do much in terms of LOC, but `llvm::interleave` with a stringstream on `subModuleName` is an option.

Reply:
In that case one should switch to a stringstream for the whole module name, I think.

Reply:
Agreed!